### PR TITLE
Add demo to deployment

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds the demo configuration from https://github.com/mrc-ide/wodin-demo-config/pull/1 to the deployment. See it in action at https://epimodels.dide.ic.ac.uk/demo/

I added some instructions as I went along, but this will change over time as we actually build a deployment strategy. I've opened mrc-3782 as something we will need to address before seriously moving forward on that.

I've moved the existing msc-idm-2022 configuration into a submodule

There's also a landing page, though perhaps at some point we could make some attempt at design (files in root/ will be served by the proxy)

Merge the other PR first, then update the submodule and proxy branch pointer in ./deploy before merge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,18 @@
 
 This is the repo for sketching out the initial deployment of the msc course to epimodels.dide.ic.ac.uk; once this is working we'll rewrite it to generally support a set of configurations.
 
-* we need to support multiple configurations easily; for now I've put the configuration in config/msc-idm-2022 to reflect one way we might achieve this practically
+* we need to support multiple configurations easily; for now I've put the configuration in `config/msc-idm-2022` to reflect one way we might achieve this practically
+* there's another configuration now at `config/demo` - this time from a submodule!
```
Comment (EmmaLRussell) [inline README.md:6]: Looks like they're both in submodules?

**Review by EmmaLRussell — COMMENTED** [review]
Ah, submodules, my favourite! :scream:

Idea of this is we can use same configs in multiple deployments?

**Comment by richfitz** [comment]
The idea behind submodules now is so that I can separate the configuration from the deployment. Once we get a deploy tool it will take a configuration of urls and map those to sites, so submodules will go away at that point.

By having different deployments in different repos, the science staff can tweak things more easily without anyone having to worry about changing someone else's stuff

File: README.md (hunk 1 above)
Comment (richfitz) [reply README.md]: I knew there was an out of date comment somewhere!

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**